### PR TITLE
smart_switch_pc_and_kies.xml

### PR DESCRIPTION
**Pull request by Tobias-B-Besemer**
Please also read my comment here...
https://github.com/az0/cleanerml/pull/171#issuecomment-472198909

...and give me a feedback that I know if my docu is wrong and need to be changed!

### CODE REVIEW

File: pending/smart_switch_pc_and_kies.xml [hunk 1]
```diff
@@ -12,53 +12,79 @@
...
   <label>Smart Switch PC and Kies</label>
-  <description>Samsung Mobiles - Smart Switch PC and Kies</description>
-  <!--One cleaner for Smart Switch PC and Kies together, because they share files.-->
+  <description>Phone management</description>
   <option id="device_error_recovery_logs">
     <label>Device Error Recovery Logs</label>
-    <description>Delete the Device Error Recovery Logs</description>
-    <action command="delete" search="glob" path="%ProgramData%\Samsung\Device Error Recovery\default-*.log"/>
+    <description>Delete the logs</description>
+    <action command="delete" search="walk.files" path="%ProgramData%\Samsung\Device Error Recovery\default-*.log"/>
```
Comment (az0) [inline pending/smart_switch_pc_and_kies.xml]: How to use `walk.all`:

* Pass only directories in the path
* Do not pass specific files
* Wildcard pattern is OK

Here it looks like you want `search="glob"` to match these log files

File: pending/smart_switch_pc_and_kies.xml [hunk 2]
```diff
@@ -12,53 +12,79 @@
...
     <action command="delete" search="walk.all" path="%ProgramData%\Samsung\DeviceProfile\Cache\"/>
-    <action command="delete" search="glob" path="%ProgramData%\Samsung\DeviceProfile\Cache\"/>
   </option>
   <option id="device_roms">
     <label>Device ROMs</label>
     <description>Delete the downloaded Device ROMs</description>
-    <action command="delete" search="walk.all" path="%LOCALAPPDATA%\Temp\Binary\"/>
-    <action command="delete" search="glob" path="%LOCALAPPDATA%\Temp\Binary\"/>
+    <action command="delete" search="walk.all" path="%Temp%\Binary\"/>
+    <action command="delete" search="file" path="%Temp%\Binary\"/>
```
Comment (az0) [inline pending/smart_switch_pc_and_kies.xml]: Lines 51-52 look ok

File: pending/smart_switch_pc_and_kies.xml [hunk 3]
```diff
@@ -12,53 +12,79 @@
...
+    <action command="delete" search="file" path="%UserProfile%\Documents\SelfMV\"/>
+  </option>
+  <option id="temp">
+    <label>Temporary files</label>
+    <description>Delete the temporary files</description>
+    <action command="delete" search="walk.all" path="%Temp%\Binary\"/>
+    <action command="delete" search="file" path="%Temp%\Binary\"/>
+    <action command="delete" search="walk.all" path="%Temp%\Samsung\"/>
+    <action command="delete" search="file" path="%Temp%\Samsung\"/>
+    <action command="delete" search="glob" path="%Temp%\SmartSwitch_*\"/>
```
Comment (az0) [inline pending/smart_switch_pc_and_kies.xml]: On line 76 do you want to delete everything under `%Temp%\SmartSwitch_*\`? It seems you want `search="walk.all"`

File: pending/smart_switch_pc_and_kies.xml (hunk 1 above)
Comment (Tobias-B-Besemer) [reply pending/smart_switch_pc_and_kies.xml]: It's `search="walk.files"`, and if wildcards works with `search="walk.all"`, it should do with `search="walk.files"`, too, or?

File: pending/smart_switch_pc_and_kies.xml (hunk 3 above)
Comment (Tobias-B-Besemer) [reply pending/smart_switch_pc_and_kies.xml]: I think it should be "walk.all" and then "glob" to delete the directories, too?

**Pull request merged**